### DISCUSSION TITLE
User defined seed for every round.

### DISCUSSION THREAD

**Discussion by dhairya0904** [ikemen-engine/Ikemen-GO#1677] [Q&A]
I want to give the same seed to every round. It works for the first round but for the subsequent rounds, with the same seed, fight results are different.
Can anyone please help me with this?
Any help is appreciated.

We are ready to compensate if anyone can help with this.

**Comment by dhairya0904** [comment]
@K4thos

**Comment by formula1** [comment]
https://github.com/ikemen-engine/Ikemen-GO/blob/develop/src/common.go#L27

Looks like setting the seed could be pretty easy

https://github.com/ikemen-engine/Ikemen-GO/blob/develop/src/script.go

I don't have much experience with lua but It might be as simple as

```go
luaRegister(l, "setRandomSeed", func(*lua.LState) int {
  Srand(int32(numArg(l, 1)))
  return 0
})
```
I'm not sure how to run this to test it but it's a start

---

A little bit more research, there are also random numbers as well as seeding in the lua scripts

Seeding Locations in the Lua scripts seem to be
- Initialized - https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/main.lua#L7
- NetPlay - https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/main.lua#L1492
- Connect to Server - https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/main.lua#L2684
- Host - https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/main.lua#L2697
- Replay File - https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/main.lua#L3372

The sszRandom seems to be coming from go (https://github.com/ikemen-engine/Ikemen-GO/blob/develop/src/script.go#L2371)

Again, I don't know much about Lua but when do you want to set the seed? You might be able to set it in response to a trigger (https://github.com/ikemen-engine/Ikemen-GO/wiki/Lua#lua-triggers)

**Comment by dhairya0904** [comment]
hi @formula1 

I can seed round 1 but not multiple rounds. I just initiated the seed in this code https://github.com/ikemen-engine/Ikemen-GO/blob/develop/src/system.go#L1981-L1994. 


```
       ------>>>> sys.randseed = 1
	for !s.endMatch {
		s.step = false
		for _, v := range s.shortcutScripts {
			if v.Activate {
				if err := s.luaLState.DoString(v.Script); err != nil {
					s.errLog.Println(err.Error())
				}
			}
		}

		// If next round
		if s.roundOver() && !fin {
                         ------>>> sys.randseed = 1
			s.round++
```

First before the game loop, I set the seed and then set the same seed when the round is restarted.

But in round 2, the same seed is giving different results. Maybe system variables that I have to reset as well.

**Comment by formula1** [comment]
@dhairya0904 

https://github.com/ikemen-engine/Ikemen-GO/blob/develop/external/script/global.lua#L205

Have you tried setting the seed in a lua script?

```lua
math.randomseed(sszRandom())
```

sszRandom seems to be a int32 so something along the lines of `math.randomseed(1)` might do the trick

https://github.com/ikemen-engine/Ikemen-GO/blob/develop/src/script.go#L2371
